src/lib/template/dom-if.html

```diff
@@ -58,7 +57,6 @@
        */
       restamp: {
         type: Boolean,
-        value: false,
```

Comment:
I would leave the default of false here. Defining a complex observer (instead of using two property observers) should prevent the execution here until the if has been set. 
Since usually restamp is not set by the user, we would treat undefined as false throughout the code. It's just a small semantic shift.